### PR TITLE
Use DynamoDB for sessions

### PR DESCRIPTION
**Pull request by jacobsimon**
**Description**:
We'd like to explore moving session stores from Redis to AWS's DynamoDB. We already use it for a few other services, but this is the first time we're exploring it with sessions. This PR essentially replaces `connect-redis` with `connect-dynamodb`. 

Before this change can be merged, the `gearman-admin-dynamo` salt branch must be merged because it contains the keys necessary for connecting to Dynamo.

**Testing**:
I've deployed to staging and confirmed that gearman-admin works and successfully writes to the gearman-admin-sessions dynamo table. I haven't been able to test the google auth flow because the redirect-urls are different from prod. 

**Deploying**:
Using connect-dynamodb means that things should "just work" but existing sessions will be lost.

### CODE REVIEW

File: .drone.yml [hunk 1]
```diff
@@ -22,11 +22,10 @@ publish:
     - $(git rev-parse --short HEAD)
     username: $$docker_username
     when:
-      branch: master
+      branch: SCHOOLS-758-use-dynamo-sessions
```
Comment (kshen0) [inline .drone.yml]: Revert before merging

File: server.coffee [hunk 2]
```diff
@@ -30,13 +25,16 @@ module.exports = (db, cb) ->
...
-        client: redis_client
-        prefix: 'gearman-admin:session:'
+      store: new DynamoDBStore
+        table: session_table
+        prefix: "gearman-admin:session:"
+        AWSRegion: process.env.AWS_DYNAMO_REGION or "us-west-1"
       cookie: httpOnly: true
       secret: process.env.SESSION_KEY or 'in development'
-      key: "gearman-admin-#{process.env.NODE_ENV}.sid"
+      key: "id"
```
Comment (kshen0) [inline server.coffee]: What was the reason behind the decision to simplify this?

**Comment by kshen0** [comment]
Looking good @jacobsimon, I just left one question

File: server.coffee (hunk 2 above)
Comment (jacobsimon) [inline server.coffee]: Meant to change that back thanks

**Comment by kshen0** [comment]
lgtm

**Pull request merged**